src/GloryDuckTables2019Lkl.cc

```diff
@@ -0,0 +1,403 @@
+    {
+      for(Int_t ibin=0;ibin<fNMasses-1;ibin++)
+	{
+	  newMassList[ibin] = fMass[ibin];
+	  newSvMinList[ibin] = fsvMin[ibin];
+	  newSvMaxList[ibin] = fsvMax[ibin];
+	}
+      delete [] fMass;
+      delete [] fsvMin;
+      delete [] fsvMax;
+    }
+
+  newMassList[fNMasses-1] = mass;
+  newSvMinList[fNMasses-1] = svmin;
+  newSvMaxList[fNMasses-1] = svmax;
+  fMass = newMassList;
+  fsvMin = newSvMinList;
+  fsvMax = newSvMaxList;
+
+  // Create and add the new ParabolaLkl object
+  ParabolaLkl* newparabola = new ParabolaLkl(npoints,sv,logL,Form("ParabolaLkl_%d",GetSampleArray()->GetEntries()));
+  AddSample(newparabola);
+  
+  SetChecked(kFALSE);
+  return 0;
+}
+
+//////////////////////////////////////////////////////////////////
+//
+// Function called by Lkl::PrintOverview
+// Print info about the experimental data
+//
+void GloryDuckTables2019Lkl::PrintData(Int_t level)
+{
+  Lkl::PrintData(level);
+
+  Margin(level); cout << "                of masses = " << fNMasses <<endl;
+  Margin(level); cout << "                 <sv>_min = " << fsvMin[0]        << " [cm^3 s^-1]" << endl;
+  Margin(level); cout << "                 <sv>_max = " << fsvMax[fNMasses-1] << " [cm^3 s^-1]" << endl;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Plot a canvas with all the -2logL, ie an array
+// of graphs (1 per mass) with -2logL vs <sv>
+// 
+// Return a pointer to the canvas, in case you want to modify it
+//
+TCanvas* GloryDuckTables2019Lkl::PlotInputData() 
+{
+  const Int_t nlines = 5;  // number of lines in parabolas canvas
+  Int_t ncols = TMath::Ceil(fNMasses/Float_t(nlines)); // number of columns in parabolas canvas
+
+  // create and divide canvas
+  TCanvas* canvas = new TCanvas("inputDataCanvas","-2logL value per energy bin and energy flux",ncols*250,nlines*250);
+  canvas->Divide(ncols,nlines);
+
+  for(Int_t ibin=0;ibin<fNMasses;ibin++)
+    {
+      ParabolaLkl* prbla   = (ParabolaLkl*)GetSample(ibin);
+      TGraph*      grprbla = prbla->GetParabola(kFALSE);
+
+      canvas->cd(ibin+1);
+      TH1I *dummypr = new TH1I(Form("dummypr_%d",ibin),Form("-2logLkl vs <sv> for m_{DM}=%.1f \\ and <sv>=[%.1f,%.1f] [cm^3 s^-1]",fMass[ibin],fsvMin[ibin],fsvMax[ibin]),1,grprbla->GetX()[1]*0.1,grprbla->GetX()[grprbla->GetN()-1]);
+      dummypr->SetDirectory(0);
+      dummypr->SetStats(0);
+      dummypr->SetXTitle("<sv> [cm^3 s^-1]");
+      dummypr->SetYTitle("#Delta(-2logL)");
+      dummypr->SetMinimum(0);
+      dummypr->SetMaximum(10);
+      dummypr->DrawCopy();
+      gPad->SetLogx();
+      delete dummypr;
+
+      // plot -2logLkl vs <sv> 
+      grprbla->Draw("l");
+      gPad->SetGrid();
+      gPad->Modified();
+      gPad->Update();
+    }
+  return canvas;
+}
+
+////////////////////////////////////////////////////////////////////////
+//
+// Set mass index to the current active one
+//
+void GloryDuckTables2019Lkl::SetActiveMass(Double_t mass)
```

Comment:
We need another function that can have the same name, but takes some Int_t index as input, and sets the active mass to fMass[index]. Only check needed is that index is non negative and  <fNMasses